tools/RAiDER/delay.py

```diff
@@ -399,14 +402,22 @@ def _build_cube_ray(
                 else:
                     pts = np.stack(pts, axis=-1)
 
+                # ray points first exist in ECEF; they are then projected to WGS84
+                # this adds slight error (order 1 mm for 500 m)
+                # at the lowest weather model layer (-500 m) the error pushes the
+                # ray points slightly below (e.g., -500.0002 m)
+                # the subsequent interpolation then results in nans
+                # here we force the lowest layer up to -500 m if it exceeds it
+                if (pts[:, :, -1] < model_zs[0]).all():
```

Comment:
@bbuzz31 I might change the implementation here slightly, using `model_zs.min()` instead of indexing, just in case there were ever to be a case where we would want to change the ordering.

Reply:
You may have to convert model_zs to a numpy array or use `np.min()` instead if it is a list.